executing/_position_node_finder.py

```diff
@@ -242,6 +242,68 @@ def fix_result(
             # keeping the old behaviour makes it possible to distinguish both cases.
 
             return node.parent
+
+        if (
+            sys.version_info >= (3, 12, 6)
+            and instruction.opname in ("GET_ITER", "FOR_ITER")
+            and isinstance(
+                node.parent.parent,
+                (ast.ListComp, ast.SetComp, ast.DictComp, ast.GeneratorExp),
+            )
+            and isinstance(node.parent,ast.comprehension)
+            and node is node.parent.iter
+        ):
+            # same as above but only for comprehensions, see:
+            # https://github.com/python/cpython/issues/123142
+
+            return node.parent.parent
+
+        if sys.version_info >= (3, 12,6) and instruction.opname == "CALL":
+            before = self.instruction_before(instruction)
+            if (
+                before is not None
+                and before.opname == "LOAD_CONST"
+                and before.positions == instruction.positions
+                and isinstance(node.parent, ast.withitem)
+                and isinstance(node.parent.parent, (ast.With, ast.AsyncWith))
+                and any(item.context_expr == node for item in node.parent.parent.items)
+            ):
+                # node positions for with-statements have change
+                # and is now equal to the expression which created the context-manager
+                # https://github.com/python/cpython/pull/120763
+
+                # with context_manager:
+                #     ...
+
+                # but there is one problem to distinguish call-expressions from __exit__()
+
+                # with context_manager():
+                #     ...
+
+                # the call for __exit__
+
+                # 20  1:5    1:22  LOAD_CONST(None)
+                # 22  1:5    1:22  LOAD_CONST(None)
+                # 24  1:5    1:22  LOAD_CONST(None)
+                # 26  1:5    1:22  CALL()         # <-- same source range as context_manager()
+
+                # but we can use the fact that the previous load for None
+                # has the same source range as the call, wich can not happen for normal calls
+
+                # we return the same ast.With statement at the and to preserve backward compatibility
+
+                return node.parent.parent
```

Comment:
```suggestion
                return node.parent
```

If we have the information, I would rather have the more specific node. If older versions don't have the info, it's OK for the behaviour to differ.

But we should test this with multiple context expressions, with aliases.

Reply:
I was thinking the same. But I don't know if I want to support different nodes for different python versions and maybe different versions of executing.

You usually assume that a `Executing.node` gives you a node of an specific type in a given context.

What do you think about introducing `Executing.exact_node` which tracks the python behavior?

There might also be cases where the source ranges become more specific than a node.
like:
``` python
a==b==c
   ^^^^
```

This is currently not the case but I think it will be someday.

Reply:
If someone wants the `With` node before 3.11 they already have to use `statements` instead of `node`. Returning a whole statement as a node just isn't very useful. And the nature of `executing` is that it's already not that stable/reliable, it has to be used carefully without assumptions. I'd rather just return the best node we can.

Reply:
Ok, but `ast.withitem` is no `expr`/`stmt` it only derives from `ast.AST` and has no line positions for example. 
Returning the context_expr of the `withitem` has the disadvantage that it is difficult to distinguish from other things like a `Call`.

I also want to provide the best node location, but I would like to wait a little bit with the decision.
Can we merge this current version which has no breaking changes and think about how we can handle these things in another issue?